discord-scripts/linear-integration.ts

```diff
@@ -0,0 +1,378 @@
+          const secret = crypto.randomBytes(32).toString("hex")
+
+          const createdWebhook = await linearClient.createWebhook({
+            url: customWebhookUrl,
+            teamId,
+            resourceTypes: ["ProjectUpdate"],
+            enabled: true,
+            label,
+            secret,
+          })
+
+          existingConnections[teamId] = {
+            webhookUrl: customWebhookUrl,
+            linearWebhookId: createdWebhook.webhook,
+            secret,
+            teamId,
+          }
+          robot.brain.set(LINEAR_BRAIN_KEY, {
+            connections: existingConnections,
+          })
+
+          await interaction.editReply({
+            content: `Connected Linear team **${team.name}** to this channel and added webhook to Linear`,
+          })
+        } else {
+          await interaction.editReply({
+            content: `Failed to generate custom webhook URL for team ${teamId}.`,
+          })
+        }
+      } else if (subcommand === DISCONNECT_SUBCOMMAND_NAME) {
+        const { channel } = interaction
+        if (!(channel instanceof TextChannel)) {
+          await interaction.reply({
+            content: "This command must be used in a text channel.",
+            ephemeral: true,
+          })
+          return
+        }
+
+        const team = await linearClient.team(teamId)
+        const sanitizedChannelName = channel.name
+          .toLowerCase()
+          .replace(/\s+/g, "-")
+        const sanitizedTeamName = team.name.toLowerCase().replace(/\s+/g, "-")
+        const label = `discord-${sanitizedChannelName}-${sanitizedTeamName}`
+
+        try {
+          const webhooks = await linearClient.webhooks()
+          const matchingWebhook = await Promise.all(
+            webhooks.nodes.map(async (webhook) => {
+              if (webhook.label !== label || !webhook.team) return null
+              const linearTeam = await webhook.team
+              return linearTeam.id === teamId ? webhook : null
+            }),
+          )
+
+          const foundWebhook = matchingWebhook.find(
+            (wh): wh is (typeof webhooks.nodes)[number] => !!wh,
+          )
+
+          if (foundWebhook) {
+            await linearClient.deleteWebhook(foundWebhook.id)
+            robot.logger.info(
+              `Deleted webhook with label ${label} (id: ${foundWebhook.id})`,
+            )
+          } else {
+            robot.logger.error(`No webhook found with label: ${label}`)
+          }
+        } catch (err) {
+          robot.logger.error(`Error deleting webhook with label ${label}:`, err)
+        }
+
+        delete existingConnections[teamId]
+        robot.brain.set(LINEAR_BRAIN_KEY, { connections: existingConnections })
+
+        await interaction.reply({
+          content: `Disconnected Linear team **${team.name}** and removed webhook from Linear.`,
+          ephemeral: true,
+        })
+      }
+    }
+  })
+  robot.router.post(
+    /^\/linear-([a-z0-9-]+)-(\d{6})$/,
```

Comment:
Let's use `/linear/` and subpath below that? Maybe even `/linear/webhook/team/...`?